### ISSUE TITLE
Project Improvements via GitHub App integration 

### ISSUE THREAD

**Issue by CHEF-KOCH** [CHEF-KOCH/CKs-FilterList#4]
I'm searching for an [implementation/app](https://github.com/marketplace) which solves the following issue via AI:

* Sort the filter lists based on regexp/alphabetically 
* Find and remove duplicates 
* Find syntax errors or line break problems or in general character issue
* Add automatic a timestamp / checksum 
* Check cosmetical filters and 'dead' or obsolete entries automatically

If someone knows a method/integration I would like to know. I theoretically could write a script on my own which would locally executed, but I like to do this via an GitHub integration in case someone provides a pull request it would be better to check this automatically which could already safe time because I would't be forced to check it locally before I merge/submit it. 

Preferable a free service but if there is only a paid solution available I need to consider if it's worth the price.  

**Edit:**
From the eMail:
* [Redundancy check for Adblock Plus rules](https://arestwo.org/famlam/redundantRuleChecker.html)
* [Web Developer Tools](https://www.browserling.com/tools)

It's a really good start I'm looking forward to check this out and might work with it.

**Comment by X8716e** [comment]
I’ll email you. It’s not an integration so I’ve not created a pull request.

**Comment by CHEF-KOCH** [comment]
Update on the subject:

* I checked all GitHub integration/apps, none of them have the requirements I need.
* There shitty scripts which require python etc which I not want, there not perfect and there require you to install dependencies.
* I got 4 emails regarding this but I must admit I not like the solutions cause there semi-optimal - thanks anyone for help anyway.

I leave it open because I'm in contact with Travis, GitHub and some others to build something like this, which might will take a while and I'm not sure if it will be free or not.
